### PR TITLE
Feat/select menu

### PR DESCRIPTION
**Pull request by cawo88**
## Proposed changes

This Pull Request adds the Select component. For desktop, we are using [react-select](https://github.com/jedwatson/react-select), and for mobile, we are using a native select menu.

## Text for the Changelog

Screenshot 2020-06-17 at 01 36 39
Screenshot 2020-06-17 at 12 56 45

### Features

- 🚀 Add the Select component using [react-select](https://github.com/jedwatson/react-select) for desktop and a native select menu for mobile.

### Bug Fixes

- 🐛 Add [rollup-plugin-svg](https://www.npmjs.com/package/rollup-plugin-svg) to import SVG for the build.
- 🐛 Add [ts-jest](https://www.npmjs.com/package/ts-jest) as a dev dependency and `jest.svgTransform.js` to transform SVG in Jest configuration. 

### Technical

- 🔧 Add [chroma-js](https://github.com/gka/chroma.js/) as a dependency.
- 🔧 Add [lodash](https://github.com/lodash/lodash) as a dependency.
- 🔧 Add [react-select](https://github.com/jedwatson/react-select) as a dependency.
- 🔧 Add [@types/chroma-js](https://www.npmjs.com/package/@types/chroma-js) as a dev dependency for chroma-js type definitions.
- 🔧 Add [@types/lodash](https://www.npmjs.com/package/@types/lodash) as a dev dependency for lodash type definitions.
- 🔧 Add [@types/react-select](https://www.npmjs.com/package/@types/react-select) as a dev dependency for react-select type definitions.
- 🌈 Refactor and replace Icon in the button component with `<DownloadIcon/>`.
- 🌈 Add `<CaretIcon/>`.
- 🗑️ Remove unused Flexbox component.
- 🌈 Add `custom-styled-components.ts` and `custom-styled-props.ts` using [styled-systems](https://styled-system.com/) in themes folder.
- 🔧 Add `isMobileDetect` utility function for mobile detection.

## Ticket

- [Create a drop-down component in the Pattern Library including styling](https://app.asana.com/0/1154394619859001/1179264451476408)

## Notes
- Will finish disabled UI styling for screen reader in another iteration.
- To test mobile, please use Developer Tools Device Toolbar mobile landscape. [Storybook Viewport Addons](https://github.com/storybookjs/storybook/tree/master/addons/viewport) has a warning message in the console and does not have user-agent detection, so did not include it until we do further investigation for device preview.
- No custom font in the pattern library.
- In mobile, the caret does not rotate on active.

### CODE REVIEW

File: src/utils/isMobileDetect.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        a.substr(0, 4),
+      )
+    ) {
+      check = true;
+    }
+  })(navigator.userAgent || navigator.vendor || window.opera);
+  return check;
+}
```
Comment (a14m) [inline src/utils/isMobileDetect.js:17]: can we convert this to typescript? and ignore the typescript violations?

File: tsconfig.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
   ],
   "include": [
     "src/**/*.ts",
-    "src/**/*.tsx"
+    "src/**/*.tsx", "src/**/*.js"
```
Comment (a14m) [inline tsconfig.json:39]: when `src/utils/isMobileDetect` is converted typescript... we don't need this line.

File: src/utils/isMobileDetect.js (hunk 1 above)
Comment (cawo88) [reply src/utils/isMobileDetect.js]: Yep, done, in this [commit](https://github.com/infographicsgroup/sapera-components/pull/26/commits/141dd6b1b9355444ac2eadefe2ba84d8bacbb6fc).

File: tsconfig.json (hunk 2 above)
Comment (cawo88) [reply tsconfig.json]: Done, in this [commit](https://github.com/infographicsgroup/sapera-components/pull/26/commits/141dd6b1b9355444ac2eadefe2ba84d8bacbb6fc).

File: tsconfig.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
   ],
   "include": [
     "src/**/*.ts",
-    "src/**/*.tsx"
+    "src/**/*.tsx",
```
Comment (a14m) [inline tsconfig.json:39]: this is invalid JSON
```suggestion
    "src/**/*.tsx"
```

**Comment by kaylarobertson3** [comment]
The dropdown looks good, and I have some comments. 

Not blocking: 

* As you mentioned in the TODO, we will keep in mind that the form elements will change when design revisits labels, placeholders, but just for a note for the next iteration, I think a floating label would work nicely here so that there is no "placeholder" text but the "choose flavor" would be the descriptive label

* I think in this case we don't need `aria-label` since we have a valid and accessible `label`. [From this article](https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/label-labelledby/), aria-label is needed for items that don't have clear UX / aren't labeled easily. In this case `label` is accessible on its own, so some screen reading programs will read out both label and aria-label

Blocking - if possible to fix easily: 

* While the focus style does match the sketch file, it's the only focus style i've seen in the designs so far, and is not accessible, so I think we should remove it. Ideally we [won't override browser default for focus styles](https://www.washington.edu/accessibility/checklist/focus/) at all, but if we do change it based on a design I think it should be tested then implemented throughout the site . It should be able to be differentiated from non-focused elements, for example, [this a11y website ](https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/label-labelledby/) has a good custom focus in their color scheme. (I guess this is mostly a note for design, but here's what i mean:) 

_current focused select dropdown in Chrome_
Screen Shot 2020-06-17 at 16 04 42

_current focused button in Chrome_
Screen Shot 2020-06-17 at 16 04 23

**Comment by cawo88** [comment]
Thanks for doing the QA @kaylarobertson3,

non-blocking
- I removed `aria-label` in this [commit](https://github.com/infographicsgroup/sapera-components/pull/26/commits/c29c04ce0e80467ded1f3b3e563afda3d269f165).

blocking
- Yep, I based the focus styling from the provided Sketch design. Here is the [commit](https://github.com/infographicsgroup/sapera-components/pull/26/commits/968bd9063ae50a7a5bb84d4e12afc0f70dcac1a9) which removes the custom styling. I will also create a ticket in the backlog later for us to customise the highlighted blue focus styling, so it looks consistent across all tab elements. For the border-radius button, we could make the focus border round as well. 
Reference: https://css-tricks.com/having-a-little-fun-with-custom-focus-styles/.

**Pull request merged**